plugins/inputs/prometheus/kubernetes.go

```diff
@@ -135,6 +144,31 @@ func podReady(statuss []*corev1.ContainerStatus) bool {
 	return true
 }
 
+func podSelector(p *Prometheus) ([]k8s.Option, error) {
+	options := []k8s.Option{}
+
+	var err error
+
+	if len(p.KubernetesLabelSelector) > 0 {
+		_, err1 := labels.Parse(p.KubernetesLabelSelector)
+		if err1 != nil {
+			return nil, fmt.Errorf("Label Selector %s validation fail. Skipping it. Error %v", p.KubernetesLabelSelector, err)
```

Comment:
Keeping with Go [error string style](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), I'd change the string to:
```
fmt.Errorf("label selector validation failed %q: %v",  p.KubernetesLabelSelector, err)
```

Reply:
will do so